src/Microsoft.Android.Build.BaseTasks/AndroidAsyncTask.cs

```diff
@@ -0,0 +1,49 @@
+using System;
+using Xamarin.Build;
+using static System.Threading.Tasks.TaskExtensions;
+
+namespace Microsoft.Android.Build.Tasks
+{
+	public abstract class AndroidAsyncTask : AsyncTask
+	{
+		/// <summary>
+		/// A helper for non-async overrides of RunTaskAsync, etc.
+		/// </summary>
+		public static readonly System.Threading.Tasks.Task Done =
+			System.Threading.Tasks.Task.CompletedTask;
+
+		public abstract string TaskPrefix { get; }
+
+		public override bool Execute ()
```

Comment:
Should this be `override sealed`, so that it can't be further overridden?

Reply:
No, because under certain circumstances we still need to override execute. One example is when calling the MSBuild GetRegisterTaskObject, that needs to be done from the UI thread.